lib/src/pgn.dart

```diff
@@ -0,0 +1,1006 @@
+                  : closeIndex;
+              _commentBuf.add(line.substring(0, endIndex));
+              _handleComment();
+              line = line.substring(closeIndex);
+              _state = ParserState.moves;
+              freshLine = false;
+            }
+          }
+      }
+    }
+  }
+
+  void _handleNag(int nag) {
+    final frame = _stack[_stack.length - 1];
+    if (frame.node != null) {
+      frame.node!.data = frame.node!.data.copyWith(nag: nag);
+    }
+  }
+
+  void _handleComment() {
+    final frame = _stack[_stack.length - 1];
+    final comment = _commentBuf.join('\n');
+    _commentBuf = [];
+    if (frame.node != null) {
+      frame.node!.data = frame.node!.data.copyWith(comment: comment);
+    } else if (frame.root) {
+      _gameComments.add(comment);
+    } else {
+      frame.startingComments ??= [];
+      frame.startingComments!.add(comment);
+    }
+  }
+}
+
+/// Default function to parse a PGN
+///
+/// Creates a list of games if multiple games found
+/// Provide a optional function [initHeaders] to create different headers other than the default
+List<Game<PgnNodeData>> parsePgn(String pgn,
+    [Headers Function() initHeaders = defaultHeaders]) {
+  final List<Game<PgnNodeData>> games = [];
+  PgnParser((Game<PgnNodeData> game, [Exception? err]) => games.add(game),
+          initHeaders)
+      .parse(pgn);
+  return games;
+}
+
+/// Parse a string for a variant if exist or return null
+Variant? parseVariant(String variant) {
+  switch (variant.toLowerCase()) {
+    case 'chess':
+    case 'chess960':
+    case 'chess 960':
+    case 'standard':
+    case 'from position':
+    case 'classical':
+    case 'normal':
+    case 'fischerandom': // Cute Chess
+    case 'fischerrandom':
+    case 'fischer random':
+    case 'wild/0':
+    case 'wild/1':
+    case 'wild/2':
+    case 'wild/3':
+    case 'wild/4':
+    case 'wild/5':
+    case 'wild/6':
+    case 'wild/7':
+    case 'wild/8':
+    case 'wild/8a':
+      return Variant.chess;
+    case 'crazyhouse':
+    case 'crazy house':
+    case 'house':
+    case 'zh':
+      return Variant.crazyhouse;
+    case 'king of the hill':
+    case 'koth':
+    case 'kingofthehill':
+      return Variant.kingofthehill;
+    case 'three-check':
+    case 'three check':
+    case 'threecheck':
+    case 'three check chess':
+    case '3-check':
+    case '3 check':
+    case '3check':
+      return Variant.threecheck;
+    case 'antichess':
+    case 'anti chess':
+    case 'anti':
+      return Variant.antichess;
+    case 'atomic':
+    case 'atom':
+    case 'atomic chess':
+      return Variant.atomic;
+    case 'horde':
+    case 'horde chess':
+      return Variant.horde;
+    case 'racing kings':
+    case 'racingkings':
+    case 'racing':
+    case 'race':
+      return Variant.racingKings;
+    default:
+      return null;
+  }
+}
+
+/// Create a [Position] from setup and variants
+Position setupPosition(Variant rules, Setup setup,
```

Comment:
not sure why we need this? in any case it does not belong here

Reply:
It is to create a position from a pgn, for example, a custom position in the analysis board. Moved the position function to the position class. If not needed, I can remove it.